src/kube-runtime/test/test_image_checker.py

```diff
@@ -0,0 +1,109 @@
+    os.path.split(os.path.realpath(__file__))[0]))
+import image_checker
+
+package_directory_com = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestImageChecker(unittest.TestCase):
+    def setUp(self):
+        try:
+            os.chdir(package_directory_com)
+        except:
+            pass
+
+    def test_image_without_username_tag(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_user_tag.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_without_tag(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_tag.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_not_exist(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_no_exist.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, False)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_local(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_local_registry.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_image_auth(self):
+        os.environ["FC_TASK_INDEX"] = "0"
+        job_path = "docker_image_auth.yaml"
+        if os.path.exists(job_path):
+            with open(job_path, 'r') as f:
+                jobconfig = yaml.load(f)
+        res = image_checker._is_docker_image_valid(jobconfig)
+        self.assertEqual(res, True)
+        del os.environ["FC_TASK_INDEX"]
+
+    def test_docker_hub_uri(self):
+        uri = "localhost/username/repo:tag"
+        self.assertFalse(image_checker._is_docker_hub_uri(uri))
+
+        uri = "username/repo:tag"
+        self.assertTrue(image_checker._is_docker_hub_uri(uri))
+
+        uri = "username/repo"
+        self.assertTrue(image_checker._is_docker_hub_uri(uri))
+
+        uri = "repo"
+        self.assertTrue(image_checker._is_docker_hub_uri(uri))
+
+        uri = "localhost:5000/repo"
+        self.assertFalse(image_checker._is_docker_hub_uri(uri))
+
+        uri = "user-name.domain/repo-0.domain:tag-0.domain"
+        self.assertTrue(image_checker._is_docker_hub_uri(uri))
```

Comment:
could you use an array to avoid duplicate code?

Reply:
fixed